pkg/core/mpt/node.go

```diff
@@ -68,3 +73,81 @@ func toBytes(n Node) []byte {
 	encodeNodeWithType(n, buf.BinWriter)
 	return buf.Bytes()
 }
+
+// MarshalJSON implements json.Marshaler.
+func (n NodeObject) MarshalJSON() ([]byte, error) {
+	return n.Node.MarshalJSON()
```

Comment:
Won't it be promoted automatically?

Reply:
There were some problems, but I can't remember or find any of them now. Removed.